tower-scripts/bin/cicd-control.sh

```diff
@@ -22,207 +25,256 @@ function on_exit() {
     fi
 }
 
-function is_running(){  
-  # Output to prevent ssh timeouts. Appears to timeout
-  # After about an hour of inactivity. 
-  while true; do
-    echo 
-    echo ".... $(date) ...."
-    sleep 600
-  done
-}
-
 trap on_exit EXIT
 
+function parse_and_set_vars() {
+  local var=$(echo $1 | awk -F= '{print $1}')
```

Comment:
don't mix scoping statements (`local`) with expressions that have return codes -- even if the right hand side expression fails the `local` will succeed and `set -o errexit` will never trigger for this line